source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,265 @@
+============================
+Snapshot Reads Specification
+============================
+
+:Spec Title: Snapshot Reads Specification (See the registry of specs)
+:Spec Version: 1.0
+:Author: Boris Dogadov
+:Advisors: Jeff Yemin, A. Jesse Jiryu Davis, Judah Schvimer
+:Status: Draft (Could be Draft, Accepted, Rejected, Final, or Replaced)
+:Type: Standards
+:Minimum Server Version: 5.0
+:Last Modified: 15-Jun-2021
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+Version 5.0 of the server introduces support for read concern level "snapshot" (non-speculative)
+for read commands outside of transactions, including on secondaries.
+This spec builds upon the `Sessions Specification <../driver-sessions.rst>`_ to define how an application
+requests "snapshot" level read concern and how a driver interacts with the server
+to implement snapshot reads.
+
+Definitions
+===========
+
+META
+----
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Terms
+-----
+
+ClientSession
+    The driver object representing a client session and the operations that can be
+    performed on it.
+
+MongoClient
+    The root object of a driver's API. MAY be named differently in some drivers.
+
+MongoCollection
+    The driver object representing a collection and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+MongoDatabase
+    The driver object representing a database and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+ServerSession
+    The driver object representing a server session.
+
+Session
+    A session is an abstract concept that represents a set of sequential
+    operations executed by an application that are related in some way. This
+    specification defines how sessions are used to implement snapshot reads.
+
+Snapshot reads
+    Reads with read concern level ``snapshot`` that occur outside of transactions on
```

Comment:
Should we add a test to assert that `snapshot` read concern level will not be appended during a transaction?

Reply:
Not sure. Now that transactions are not allowed with snapshot reads, and we got a test for that. I don't see how snapshot readconcern can end up being appended in transaction.

Reply:
Yes I think this is covered by the new "StartTransaction fails in snapshot session" test; thanks.